### PR TITLE
Ontology development

### PR DESCRIPTION
**Pull request by vandungdo**
After a lot of tries, I finally fixed it (it worked perfectly when I run it locally with pydoit, but it showed an error on GitHub action and I don't know why @@). The reason is the rdflib version.

All steps in the workflow now (minimum working example workflow) went through, and are ready to be merged (I hope). It seems the test calibration workflow is not related to our work now, so I decided to delete it.

After merging, we can delete the ontology_development branch. And starting to create issues and branches...

### CODE REVIEW

File: usecases/Concrete/dodo.py [hunk 1]
```diff
@@ -38,19 +39,33 @@ def task_emodul():
             'basename': 'create processeddata folder',
             'actions': ['mkdir %s ' % emodulProcesseddataFolder]
         }
+        yield {
```
Comment (joergfunger) [inline usecases/Concrete/dodo.py:42]: file_dep are missing

File: usecases/Concrete/knowledgeGraph/emodul/emodul_metadata_extraction.py [hunk 2]
```diff
@@ -96,42 +97,73 @@ def eModul_metadata(filePath, fileName ):
...
-    'length': column_data(metadata, operatorIndex, 'Bediener Information', 'Lnge'),
-    'data collection time': column_data(metadata, dataCollectionIndex, 'Datenerfassung', 'Zeit'),
-    'data collection timestamp': column_data(metadata, dataCollectionIndex, 'Datenerfassung', 'Zeitpunkt')
-})
-
-dataFrame.to_csv(os.path.join(baseDir0,'E-modul-processed-data/emodul_metadata.csv'), index = False)
-
-print(dataFrame.head())
\ No newline at end of file
+def generate_metadata_yaml_file_from_emodul_BAM(locationOfRawData, locationOfMetaData):
```
Comment (joergfunger) [inline usecases/Concrete/knowledgeGraph/emodul/emodul_metadata_extraction.py:108]: I think the return function should rather be the dict, that you can then store in a separate file if needed (this way you can test the functionality without intermediate files in pytest)

File: usecases/Concrete/knowledgeGraph/emodul/emodul_metadata_extraction.py [hunk 3]
```diff
@@ -96,42 +97,73 @@ def eModul_metadata(filePath, fileName ):
...
+    metadataDict = [{
+        'experiment raw name': [i[experimentNameIndex]['experimentName'] for i in metadata],
+        'data type': [i[dataTypeIndex]['data type'] for i in metadata],
+        'operator time': column_data(metadata, operatorIndex, 'Bediener Information', 'Zeit'),
+        'operator timestamp': column_data(metadata, operatorIndex, 'Bediener Information', 'Zeitpunkt'),
+        'operator date': column_data(metadata, operatorIndex, 'Bediener Information', 'Datum'),
+        'tester': column_data(metadata, operatorIndex, 'Bediener Information', 'Prfer'),
+        'sample name': column_data(metadata, operatorIndex, 'Bediener Information', 'Probenbezeichnung'),
+        'remark': column_data(metadata, operatorIndex, 'Bediener Information', 'Bemerkungen'),
+        'weight': column_data(metadata, operatorIndex, 'Bediener Information', 'Masse'),
```
Comment (joergfunger) [inline usecases/Concrete/knowledgeGraph/emodul/emodul_metadata_extraction.py:139]: if appropriate, assumptions on the units should be stored in the metadata dict as well

File: usecases/Concrete/knowledgeGraph/emodul/emodul_test.py [hunk 4]
```diff
@@ -5,8 +5,10 @@
 print('input: name of the emodul expriment')
 print('output: processed data of that experiment and the specimen parameters')
 nameOfExperiment = 'Werner 7.0 M IV E-Modul 28d v. 06.08.14 Probe 6'
+print(input_emodul_data_for_calibration(nameOfExperiment)['processedDataPath'])
```
Comment (joergfunger) [inline usecases/Concrete/knowledgeGraph/emodul/emodul_test.py:8]: remove auxiliary prints

File: usecases/Concrete/knowledgeGraph/emodul/emodul_validation.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+SCHEMA = Namespace('http://schema.org/')
+
+"""
+Given a path to a shacl shape and a path to an rdf file, this function tests the rdf data against the specified shacl shapes.
+The result is an rdflib graph containing the validation report, if it is empty the validation was successful.
+"""
+def test_graph(rdf_graph: Graph, shapes_graph: Graph) -> Graph:
+
+    conforms, result_graph, _ = validate(
```
Comment (joergfunger) [inline usecases/Concrete/knowledgeGraph/emodul/emodul_validation.py:24]: why does validate return a result graph that is then further used? I would have expected that rdf_graph conforms, then the return value is true, or it does not, then it returns an error with the corresponding things that fail.

File: usecases/Concrete/knowledgeGraph/mixture/extractMetadataFromMischung.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    for deutsch, english in translation.itertuples(index=False,name=None):
+        df = df.replace(deutsch,english,regex=True)
+    return df
+    
+# Set up lists to fill inside the loop and build final dataframe from
+list_of_dfs = []
+list_of_excelsheets = []
+
+# main part: extract the metadata for each excelfile and his sheet(s) in the folder
+for excelsheet in excelsheetlist:
```
Comment (joergfunger) [inline usecases/Concrete/knowledgeGraph/mixture/extractMetadataFromMischung.py:41]: Still not a function, this loop is done in pydoit by definining the dependencies e.g. [here](https://github.com/BAMresearch/ModelCalibration/blob/d60dd8e001e927050ab27907e7be282fe17b49fc/usecases/linearFunction/dodo.py#L29)

File: usecases/Concrete/knowledgeGraph/shape.ttl [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix mseo: <https://purl.matolab.org/mseo/mid/> .
+@prefix cco: <http://www.ontologyrepository.com/CommonCoreOntologies/> .
+@prefix obo: <http://purl.obolibrary.org/obo/> .
+@prefix con: <http://w3id.org/concrete> .
+@prefix tt: <http://w3id.org/concrete/youngs> .
+
+# for a documentation of shacl see https://www.w3.org/TR/shacl/
```
Comment (joergfunger) [inline usecases/Concrete/knowledgeGraph/shape.ttl:22]: this is a specific shacle shape and should be named accordingly (not shape.ttl)

**Review by joergfunger — CHANGES_REQUESTED** [review]
Not fully sure what is supposed to be adressed in this merge request, and what is supposed to be done afterwards.

**Review by joergfunger — APPROVED** [review]

**Comment by joergfunger** [comment]
@vandungdo What is the current status? Can we merge, or what would be a timeline?

**Comment by vandungdo** [comment]
> @vandungdo What is the current status? Can we merge, or what would be a timeline?

The branch is now ready to be merged. Please ask @soudehMasoudian for adding her unit test script for each step in the new folder structure. Otherwise, we can merge it first and she needs to create new branch for testing

**Pull request merged**